eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -8,6 +8,9 @@ parameters:
   - name: ServiceDirectory
     type: string
     default: not-specified
+  - name: TestProxy
```

Comment:
you are not passing this parameter ci.yml. We need to pass this parameter down to test.yml template.  Here [ci.yml](https://github.com/Azure/azure-sdk-for-js/pull/18466/files#diff-8846d7741afb575539571d7b46758247229daaa4336751c6b3e743fd284426faR48)

and https://github.com/Azure/azure-sdk-for-js/blob/main/eng/pipelines/templates/jobs/ci.yml#L78

and https://github.com/Azure/azure-sdk-for-js/blob/main/eng/pipelines/templates/jobs/ci.tests.yml#L54

Reply:
Does it look good now?

Reply:
yeah. seems reasonable. now let's see what breaks 👍